src/platform/OpenThread/GenericThreadBorderRouterDelegate.h

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app/clusters/thread-border-router-management-server/thread-br-delegate.h>
+#include <inet/IPAddress.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace ThreadBorderRouterManagement {
+
+class GenericOpenThreadBorderRouterDelegate : public Delegate
+{
+public:
+    static constexpr char kThreadBorderRourterName[]      = "OpenThread-ThreadBR";
```

Comment:
Should we also add a way to change the default name?

Reply:
Added the name input in constructor.